### PR TITLE
Shiv/pagination 2

### PR DESCRIPTION
**Pull request by parikhshiv**

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -150,18 +150,45 @@ exports.translations = functions.https.onRequest(async (req, res) => {
 // todo(parikhshiv) - made this method mainly for development, can be
 // replaced / expanded
 exports.getEntireCollection = functions.https.onRequest(async (req, res) => {
+  const pageSize = +req.query.pageSize;
+  const pageNum = +req.query.pageNum;
+  const state = req.query.state;
+  const needsRecording = req.query.needsRecording;
   return cors(req, res, async () => {
-    const collection = admin.firestore().collection(req.query.collectionName);
+    let collection = admin.firestore().collection(req.query.collectionName)
```
Comment (borismus) [inline functions/index.js]: I think this approach may actually fetch everything from firestore, and then do filtering in code. This seems like it'd be pretty slow. Could be wrong tho?

This seems like the better way of doing it: https://firebase.google.com/docs/firestore/query-data/query-cursors

File: src/common/filters/FilterChips.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.filter-chips {
+  margin-bottom: 24px;
+}
+
+.filter-chip.MuiChip-root {
```
Comment (borismus) [inline src/common/filters/FilterChips.css]: Better not to depend on MUI class names.

File: functions/index.js (hunk 1 above)
Comment (parikhshiv) [reply functions/index.js]: Right, looked into this, and unfortunately pagination in firebase isn't very intuitive. We're not able to start at a specific numeric index in a query, only at a specific value, so we'd need to store the last word on the "previous" page in order to get to the "next" page. I did some performance testing, and with ~1000 words, this call takes less than a second to complete. Could be better, but I'll note that the current version of the app takes about 7 seconds to render (mainly due to the browser rendering time), and this PR cuts that down significantly.

File: src/common/filters/FilterChips.css (hunk 2 above)
Comment (parikhshiv) [reply src/common/filters/FilterChips.css]: fixed.

File: functions/index.js (hunk 1 above)
Comment (borismus) [reply functions/index.js]: Alright, I suppose we can revisit this if/when the number of entries in the database is so large that it becomes a performance issue.

**Pull request merged**